src/test/java/org/apache/sling/graphql/core/engine/DefaultQueryExecutorTest.java

```diff
@@ -188,4 +191,30 @@ public void unionFetcherTest() throws Exception {
         assertThat(json, hasJsonPath("$.data.unionFetcher.items[0].testingArgument", equalTo("1, 2, 3")));
         assertThat(json, hasJsonPath("$.data.unionFetcher.items[1].path", equalTo(resource.getPath())));
     }
+
+    @Test
+    public void selectionSetTest() throws Exception {
```

Comment:
This only covers unions. I would suggest to add a test for "normal" nested structures as well, as it would also be interesting how they would be reflected in the output, mostly if the type name would be included as well or if that's only true for `union`s. Also it would be interesting for the latter case to determine if an item of the selection set determines a union differentiator or actual data, as we couldn't simply follow the result data in the fetcher implementation in this case.

Reply:
I will add a that in the test, only "inline" so currently in our cases Union would have the "type" as the name of the field.